File: employee-management-frontend/src/services/EmployeeService.js
import axios from 'axios';

// Base URL of the Spring Boot Backend API
const API_BASE_URL = 'http://localhost:8080/api/employees';

class EmployeeService {
    
    /**
     * Fetch all employees from the backend.
     * Route: GET http://localhost:8080/api/employees
     */
    getAllEmployees() {
        return axios.get(API_BASE_URL);
    }

    /**
     * Create/Add a new employee.
     * Route: POST http://localhost:8080/api/employees
     * @param {Object} employeeDto - Employee details
     */
    createEmployee(employeeDto) {
        return axios.post(API_BASE_URL, employeeDto);
    }

    /**
     * Fetch details of a single employee by their ID.
     * Route: GET http://localhost:8080/api/employees/{id}
     * @param {number} employeeId
     */
    getEmployeeById(employeeId) {
        return axios.get(`${API_BASE_URL}/${employeeId}`);
    }

    /**
     * Update an existing employee.
     * Route: PUT http://localhost:8080/api/employees/{id}
     * @param {number} employeeId
     * @param {Object} employeeDto - Updated employee details
     */
    updateEmployee(employeeId, employeeDto) {
        return axios.put(`${API_BASE_URL}/${employeeId}`, employeeDto);
    }

    /**
     * Delete an employee record.
     * Route: DELETE http://localhost:8080/api/employees/{id}
     * @param {number} employeeId
     */
    deleteEmployee(employeeId) {
        return axios.delete(`${API_BASE_URL}/${employeeId}`);
    }
}

// Export an instance of the class for direct import and usage across pages
export default new EmployeeService();

File: employee-management-backend/src/main/java/com/example/employeemanager/service/EmployeeService.java
package com.example.employeemanager.service;

import com.example.employeemanager.dto.EmployeeDto;
import java.util.List;

/**
 * Service Interface outlining the business logic operations.
 * Declares the standard CRUD methods.
 */
public interface EmployeeService {
    
    /**
     * Create and store a new employee.
     * 
     * @param employeeDto DTO object representing the new employee.
     * @return EmployeeDto The saved employee, containing the generated database ID.
     */
    EmployeeDto createEmployee(EmployeeDto employeeDto);

    /**
     * Retrieve a single employee by their unique ID.
     * 
     * @param employeeId The database ID of the employee.
     * @return EmployeeDto The mapped DTO of the employee.
     */
    EmployeeDto getEmployeeById(Long employeeId);

    /**
     * Retrieve all employee records from the database.
     * 
     * @return List<EmployeeDto> A list of all employees in DTO format.
     */
    List<EmployeeDto> getAllEmployees();

    /**
     * Update details of an existing employee.
     * 
     * @param employeeId The ID of the employee to update.
     * @param updatedEmployeeDto The DTO representing the updated fields.
     * @return EmployeeDto The updated employee database entry in DTO format.
     */
    EmployeeDto updateEmployee(Long employeeId, EmployeeDto updatedEmployeeDto);

    /**
     * Delete an employee record from the database.
     * 
     * @param employeeId The database ID of the employee to be deleted.
     */
    void deleteEmployee(Long employeeId);
}
